Source/SwiftLintFramework/Rules/Metrics/LargeTupleRule.swift

```diff
@@ -24,152 +14,29 @@ public struct LargeTupleRule: ASTRule, ConfigurationProviderRule {
-                let reason = "Tuples should have at most \(configuration.warning) members."
-                return StyleViolation(ruleDescription: Self.description,
-                                      severity: parameter.severity,
-                                      location: Location(file: file, byteOffset: location),
-                                      reason: reason)
-            }
-
-            return nil
-        }
-    }
-
-    private func violationOffsetsForTypes(in file: SwiftLintFile, dictionary: SourceKittenDictionary,
-                                          kind: SwiftDeclarationKind) -> [(offset: ByteCount, size: Int)] {
-        let kinds = SwiftDeclarationKind.variableKinds.subtracting([.varLocal])
-        guard kinds.contains(kind),
-            let type = dictionary.typeName,
-            let offset = dictionary.offset else {
-                return []
-        }
-
-        let sizes = violationOffsets(for: type).map { $0.1 }
-        return sizes.max().flatMap { [(offset: offset, size: $0)] } ?? []
-    }
-
-    private func violationOffsetsForFunctions(in file: SwiftLintFile, dictionary: SourceKittenDictionary,
-                                              kind: SwiftDeclarationKind) -> [(offset: ByteCount, size: Int)] {
-        let contents = file.stringView
-        guard SwiftDeclarationKind.functionKinds.contains(kind),
-            let returnRange = returnRangeForFunction(dictionary: dictionary),
-            let returnSubstring = contents.substringWithByteRange(returnRange) else {
-                return []
-        }
-
-        let offsets = violationOffsets(for: returnSubstring, initialOffset: returnRange.location)
-        return offsets.sorted { $0.offset < $1.offset }
-    }
-
-    private func violationOffsets(for text: String, initialOffset: ByteCount = 0) -> [(offset: ByteCount, size: Int)] {
-        guard let ranges = try? parenthesesRanges(in: text) else {
-            return []
-        }
-
-        var text = text.bridge()
-        var offsets = [(offset: ByteCount, size: Int)]()
-
-        for (range, kind) in ranges {
-            let substring = text.substring(with: range)
-            if kind != .generic,
-                let byteRange = StringView(text).NSRangeToByteRange(start: range.location, length: range.length),
-                !containsReturnArrow(in: text.bridge(), range: range) {
-                let size = substring.components(separatedBy: ",").count
-                let offset = byteRange.location + initialOffset
-                offsets.append((offset: offset, size: size))
-            }
-
-            let replacement = String(repeating: " ", count: substring.bridge().length)
-            text = text.replacingCharacters(in: range, with: replacement).bridge()
-        }
+    public func validate(file: SwiftLintFile) -> [StyleViolation] {
+        let visitor = LargeTupleRuleVisitor()
+        return visitor
```

Comment:
Nit: can use implicit return here

```suggestion
        LargeTupleRuleVisitor()
```

Up to you though, strictly a stylistic choice.